File: Examples/test-suite/php/cpp11_auto_variable_runme.php
<?php

require "tests.php";

// No new functions
check::functions(array());
check::classes(array('cpp11_auto_variable'));
check::globals(array('f', 't', 'zero', 'one', 'la', 'da', 'fa', 'lc', 'dc', 'fc', 'pi_approx', 'wstring_lit_len1', 'wstring_lit_len2', 'Bar', 'Bar2', 'Bar3', 'Foo', 'Foo2', 'Foo3', 'NOEXCEPT_FUNC'));

check::equal(f_get(), false);
check::equal(gettype(f_get()), "boolean");
check::equal(t_get(), true);
check::equal(gettype(t_get()), "boolean");

t_set(false);
check::equal(t_get(), false);

check::equal(function_exists('f_set'), false, "f should be constant but f_set() exists");

check::equal(zero_get(), 0);
check::equal(gettype(zero_get()), "integer");
check::equal(one_get(), 1);
check::equal(gettype(one_get()), "integer");

zero_set(42);
check::equal(zero_get(), 42);

check::equal(function_exists('one_set'), false, "one should be constant but one_set() exists");

check::equal(fa_get(), 1.0);
check::equal(da_get(), 1.0);
// PHP doesn't have a native "long double" type, so SWIG/PHP doesn't have
// typemaps for it and so it should get wrapped as an opaque type.
check::str_contains(la_get(), "SWIGPointer(");

check::equal(fc_get(), 1.0);
check::equal(dc_get(), 1.0);
// PHP doesn't have a native "long double" type, so SWIG/PHP doesn't have
// typemaps for it and so it should get wrapped as an opaque type.
check::str_contains(lc_get(), "SWIGPointer(");

check::equal(wstring_lit_len1_get(), 6);
check::equal(wstring_lit_len2_get(), 6);

File: Examples/test-suite/lua/cpp11_auto_variable_runme.lua
local v=require("cpp11_auto_variable")



assert(not v.f)
assert(swig_type(v.f) == "boolean")
assert(v.t)
assert(swig_type(v.t) == "boolean")

v.t = false
assert(not v.t)

assert(v.zero == 0)
assert(swig_type(v.zero) == "number")
assert(v.one == 1)
assert(swig_type(v.one) == "number")

v.zero = 42
assert(v.zero == 42)

assert(v.fa == 1)
assert(v.da == 1)
assert(swig_type(v.la) == "long double *") -- swig do not support long double

assert(v.fc == 1)
assert(v.dc == 1)
assert(swig_type(v.lc) == "long double *") -- swig do not support long double

assert(v.wstring_lit_len1 == 6)
assert(v.wstring_lit_len2 == 6)
assert(v.pi_approx == (355.0 / 133.0)) -- Lua uses same C++ double math!
assert(v.Bar ==  "foo")
assert(v.Foo ==  "bar")
assert(v.Bar2 == "foo")
assert(v.Foo2 == "bar")
assert(v.Bar3)
assert(v.Foo3 == 1)
assert(v.CAST_HAD_WRONG_PRECEDENCE1 == 2)
assert(v.CAST_HAD_WRONG_PRECEDENCE2 == 2)
assert(v.CAST_HAD_WRONG_PRECEDENCE3 == -1)
assert(v.CAST_HAD_WRONG_PRECEDENCE4 == 3)
-- NOEXCEPT_FUNC TODO should be pointer to func()
